src/huggingface_hub/inference_api.py

```diff
@@ -0,0 +1,108 @@
+import json
+import logging
+from typing import Optional
+
+import requests
+
+from .hf_api import HfApi, HfFolder
+from .validation import ALL_TASKS, check_inputs, check_params
+
+
+logger = logging.getLogger(__name__)
+
+
+ENDPOINT = "https://api-inference.huggingface.co/pipeline"
+
+
+class InferenceApi:
+    """Client to configure requests and make calls to the HuggingFace Inference API.
+
+    Example:
+
+            >>> from huggingface_hub.inference_api import InferenceApi
+
+            >>> # Mask-fill example
+            >>> api = InferenceApi("bert-base-uncased")
+            >>> api.set_inputs(inputs="The goal of life is [MASK].")
+            >>> api.call()
+
+            >>> # Question Answering example
+            >>> api = InferenceApi("deepset/roberta-base-squad2")
+            >>> api.set_inputs(question="What's my name?", context="My name is Clara and I live in Berkeley.")
+            >>> api.call()
+
+            >>> # Zero-shot example
+            >>> api = InferenceApi("typeform/distilbert-base-uncased-mnli")
+            >>> api.set_inputs(inputs="Hi, I recently bought a device from your company but it is not working as advertised and I would like to get reimbursed!")
+            >>> api.set_params(candidate_labels=["refund", "legal", "faq"])
+            >>> api.call()
+    """
+
+    def __init__(
+        self, repoId: str, task: Optional[str] = None, gpu: Optional[bool] = False
```

Comment:
should probably take an optional param `use_auth_token` like other HfApi methods

Reply:
Done, although I'm adding `token` since we'll always need a token.